src/mower_logic/src/monitoring/monitoring.cpp

```diff
@@ -125,114 +113,119 @@ void pose_received(const xbot_msgs::AbsolutePose::ConstPtr &msg) {
   xbot_msgs::SensorDataDouble sensor_data;
   sensor_data.stamp = msg->header.stamp;
   sensor_data.data = msg->position_accuracy;
-  gps_accuracy_data_pub.publish(sensor_data);
+
+  auto sc_it = sensor_configs.find("om_gps_accuracy");
+  if (sc_it != std::end(sensor_configs)) {
+    sc_it->second.data_pub.publish(sensor_data);
+  }
+}
+
+/**
+ * Set sensor limits, dependent on sensor type.
+ * Determine limits by config settings or param file.
+ */
+void set_sensor_limits(SensorConfig &sensor_config) {
```

Comment:
Maybe it would make sense to create small callback functions for the 7 variants and reference them from the `sensor_configs` map. Then you don't have to look for matching type/name with lots of conditionals, but can jump to the implementation right from the map entry.

Reply:
Yeees, quite cool idea! Now that there's one common sensor_config I should use that for the limits too.